scripts/upgrade

```diff
@@ -70,10 +78,29 @@ if [ "$upgrade_type" == "UPGRADE_APP" ]
 then
 	ynh_script_progression --message="Upgrading source files..." --weight=6
 
+# Create a temporary directory
+  tmpdir="$(mktemp -d)"
+
+    # Backup the config file in the temp dir
+  	cp -a "$final_path/config.php" "$tmpdir/config.php"
+
+	# Remove the app directory securely
+	ynh_secure_remove --file="$final_path"
+
 	# Download, check integrity, uncompress and patch the source from app.src
 	ynh_setup_source --dest_dir="$final_path"
+
+	# Copy the admin saved settings from tmp directory to final path  
+  	cp -a "$tmpdir/config.php" "$final_path/config.php" 
+
+  	# Remove the tmp directory securely
+  	ynh_secure_remove --file="$tmpdir"
```

Comment:
I'm not sure but you may be able to simplify this with (assuming Yunohost >= 4.2) :

```suggestion
	# Download, check integrity, uncompress and patch the source from app.src
	ynh_setup_source --dest_dir="$final_path" --keep="$final_path/config.php"
```

Reply:
I have tried, but the app seems to complain about mixed version files...